### ISSUE TITLE
[Feature Request] - Workspaces

### ISSUE THREAD

**Issue by nawlbergs** [TablePlus/TablePlus#3114]
# TLDR;
- Add workspace menu on left (maybe like slack) 
- Right click a table -> Add to workspace.
- When in a workspace, would filter all tables to only tables you've added to the workspace.

# Detailed
We have 100's of tables in our application.  Working on a feature requires dozens of tables... all named odd/different etc...  TablesPlus is open on my computer 100% of the day as I develop/etc.   I am constantly flipping back and forth between tables/filter/pinning etc in order to focus on the tables/data I need to view depending on what Features im working on within our application.

It would be awesome to be able to create a Workspace (or group or something), add/remove tables to that workspace... so you can remove all the clutter. 

Thanks.

**Comment by nawlbergs** [comment]
I just realized your active "window" is called a workspace... so maybe a different name... either way the feature request remains valid.  :)

**Comment by huyphams** [comment]
Maybe add custom group... would help? Also you can navigate quickly using Command + P
<img width="544" alt="Screenshot 2023-10-16 at 5 04 48 PM" src="https://github.com/TablePlus/TablePlus/assets/3654622/dc19d77f-4e4b-412d-9c21-d946232ad9cc">

**Comment by landovsky** [comment]
I have an issue with workspace that it does not behave like one - you can't save it or restore it. When you close TablePlus and open new workspace and database, you can pseudo restore by opening new tabs that will contain whatever they contained before, but:
- it would not open saved .sql files you had opened before
- when app crashes, you can lose everything

If you just mimic how VS Code behaves when you close / reopen a project, it would be just fine.

This is such a basic feature when working regularly on projects, I don't understand how other users are working around its absence.
